app/pods/components/agenda/agendaitem/agendaitem-case/subcase-documents/component.js

```diff
@@ -13,327 +9,312 @@ import {
 import moment from 'moment';
 import config from 'fe-redpencil/utils/config';
 
-import VRDocumentName from 'fe-redpencil/utils/vr-document-name';
-
-export default Component.extend(
-  {
-    currentSession: inject(),
-    classNames: ['vl-u-spacer--large'],
-
-    store: service(),
-    documentsInCreation: A([]), // When creating new documents
-
-    isAddingNewDocument: false,
-    isEditing: false,
-    isLoading: false,
-    shouldShowLinkedDocuments: true,
-    item: null,
-    documentsToLink: A([]),
-    model: alias('uploadedFiles'),
-
-    document: deprecatingAlias('documentContainer', {
-      id: 'model-refactor.documents',
-      until: '?',
-    }),
-    documentContainer: null, // When adding a new version to an existing document
-    defaultAccessLevel: null, // when creating a new document
+import { tracked } from '@glimmer/tracking';
+
+export default class SubcaseDocuments extends Component {
+  @service currentSession;
+  @service store;
+  @tracked isEditing = false;
+  @tracked isLoading = false;
+  @tracked isAddingNewDocument = false;
+  @tracked shouldShowLinkedDocuments = true;
+  @tracked documentsInCreation = A([]); // When creating new documents
+  @tracked defaultAccessLevel = null;
+  @tracked isLinkingOldDocument = false;
+  @tracked documentsToLink = A([]);
+
+  documentContainer = null;
+
+  constructor() {
+    super(...arguments);
+    this.store.query('document-type', {
+      sort: 'priority', 'page[size]': 50,
+    }).then((types) => {
+      this.documentTypes = types;
+    });
+  }
 
+  get governmentCanViewDocuments() {
+    const isAgendaItem = this.args.item.get('modelName') === 'agendaitem';
+    const isSubcase = this.args.item.get('modelName') === 'subcase';
+    const isOverheid = this.currentSession.isOverheid;
 
-    get overheidCanViewDocuments() {
-      const isAgendaItem = this.item.get('modelName') === 'agendaitem';
-      const isSubcase = this.item.get('modelName') === 'subcase';
-      const isOverheid = this.currentSession.isOverheid;
+    if (isAgendaItem) {
+      const documentsAreReleased = this.args.item.get('agenda.createdFor.releasedDocuments');
+      return !(isOverheid && !documentsAreReleased);
+    }
 
-      if (isAgendaItem) {
-        const documentsAreReleased = this.item.get('agenda.createdFor.releasedDocuments');
-        return !(isOverheid && !documentsAreReleased);
-      }
-
-      if (isSubcase) {
-        const documentsAreReleased = this.item.get('requestedForMeeting.releasedDocuments');
-        return !(isOverheid && !documentsAreReleased);
-      }
+    if (isSubcase) {
+      const documentsAreReleased = this.args.item.get('requestedForMeeting.releasedDocuments');
```

Comment:
Dit lijkt me ook een relatie, moeten we dit niet awaiten?

Reply:
Neen omdat dit in een 'get' gebeurd gaat die opnieuw getriggerd worden het moment dat de relatie is binnengehaald